lib/ansible/cli/galaxy.py

```diff
@@ -1396,8 +1397,10 @@ def execute_list_collection(self, artifacts_manager=None):
 
         output_format = context.CLIARGS['output_format']
         collections_search_paths = set(context.CLIARGS['collections_path'])
+        if context.CLIARGS['basedir']:
+            collections_search_paths.update([context.CLIARGS['basedir']])
         collection_name = context.CLIARGS['collection']
-        default_collections_path = AnsibleCollectionConfig.collection_paths
+        default_collections_path = AnsibleCollectionConfig.collection_paths + [context.CLIARGS['basedir']]
```

Comment:
you want to put basedir first to match normal precedence

Reply:
Alright

Reply:
Quick question....

from what you already said, collections work on a first found basis
With that in mind, shoudn't the list of paths from
```python
from ansible.utils.collection_loader import AnsibleCollectionConfig
# ...
AnsibleCollectionConfig.collection_paths
```

Always return a list with the same order?

Reply:
yes, it should, lists preserve order in Python

Reply:
Then we might have another issue... I'll try to reproduce what I did last night and open a new issue so that if i am doing something wrong it can be pointed out ;-)

Reply:
note that collections_search_paths is a 'set' which does NOT preserve orde, but default_collections_path should